src/zmq/zmqpublishnotifier.cpp

```diff
@@ -250,7 +249,7 @@ bool CZMQPublishRawBlockNotifier::NotifyBlock(const CBlockIndex *pindex)
     const Consensus::Params& consensusParams = Params().GetConsensus();
     CDataStream ss(SER_NETWORK, PROTOCOL_VERSION | RPCSerializationFlags());
     CBlock block;
-    if (!ReadBlockFromDisk(block, pindex, consensusParams)) {
+    if (!m_node.GetBlockManager().ReadBlockFromDisk(block, pindex, consensusParams)) {
```

Comment:
nit: Seems unnecessary to pass the entire `NodeContext` when you only need the blockman.

Feel free to ignore, but would be interested in why you chose to do it this way.

Reply:
The blockman and chainstate instances may be deleted during the lifetime of the zmq instance similarly as described in mzsumsande's [comment](https://github.com/bitcoin/bitcoin/pull/27125#discussion_r1160180267) when iterating the for loop. If we pass a reference to the `NodeContext` instead, we have better guarantees that its blockman reference points to an actual object.

Reply:
Is that actually possible? Seems like all the validation interfaces including the zmq one are unregistered before chainman is destroyed?

So it can't be holding a reference to blockman that is actually destroyed, see:

https://github.com/bitcoin/bitcoin/blob/f3f5c9712670fc2c9fb3b57580146aa248a5d36f/src/init.cpp#L327-L343

Or are you just saying that having a reference to the node context has better lifetime guarantees because basically nothing outlives the node context?

Reply:
> Is that actually possible? 

Likely that I am missing something in the logic. The way I see it each iteration of the for loop creates and destroys the `chainman` by calling `std::make_unique` again, while the `Shutdown` code is not called during each iteration.